torch/lib/THCUNN/generic/LookupTable.cu

```diff
@@ -172,27 +173,30 @@ void THNN_(LookupTable_renorm)(
   int64_t stride = THCTensor_(stride)(state, weight, 0);
 
   // get the unique indices
-  thrust::device_ptr<real> weight_ptr(THCTensor_(data)(state, weight));
   thrust::device_ptr<THCIndex_t> idx_ptr(THCIndexTensor_(data)(state, idx));
   thrust::device_ptr<THCIndex_t> end_ptr(thrust::unique(idx_ptr, idx_ptr+numel));
   numel = end_ptr - idx_ptr;
 
-  pow_v<real, accreal> unary_pow(normType);
-  thrust::plus<accreal> binary_plus;
-  // numel << stride, since idx usually contains sparse row indices
-  for (THCIndex_t i = 0; i < numel; i++)
-  {
-    THCIndex_t k = idx_ptr[i] - TH_INDEX_BASE;
-    thrust::device_ptr<real> row_ptr = weight_ptr + k * stride;
-    accreal norm = thrust::transform_reduce(row_ptr, row_ptr + stride,
-      unary_pow, (accreal)0, binary_plus);
-    norm = std::pow(norm, (accreal) (1.0 / normType));
-    if (norm > ScalarConvert<real, accreal>::to(maxNorm))
-    {
-      multiply_s<real> unary_mul(ScalarConvert<accreal, real>::to(maxNorm / (norm + 1e-7)));
-      thrust::transform(row_ptr, row_ptr + stride, row_ptr, unary_mul);
-    }
-  }
+  real * weight_ptr_raw = THCTensor_(data)(state, weight);
+  THCIndex_t * idx_ptr_raw = THCIndexTensor_(data)(state, idx);
+
+  THCIndex_t num_blocks = numel;
+  // get the next highest power of 2 for algo correctness
+  int64_t threads_per_block = stride - 1;
```

Comment:
This won't work, there's an upper bound on the number of threads per block that is based on the CUDA compute capability, typically 512 or 1024.

I would just pick a fixed block size, something like 128 or 256.